api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -108,16 +81,22 @@ public File getOutputFileToWrite(Obs obs) throws IOException {
 	 * 
 	 * @param filename
 	 * @return the filepart after the period in the given filename
+	 * @deprecated since 2.1.3 use {@link org.apache.commons.io.FilenameUtils#getExtension(String)}
+	 *             instead.
 	 */
+	@Deprecated
 	public String getExtension(String filename) {
-		String[] filenameParts = filename.split("\\.");
-		
-		log.debug("titles length: " + filenameParts.length);
+		String result = FilenameUtils.getExtension(filename);
 		
-		String extension = (filenameParts.length < 2) ? filenameParts[0] : filenameParts[filenameParts.length - 1];
-		extension = StringUtils.isNotEmpty(extension) ? extension : "raw";
+		if (StringUtils.isEmpty(result)) {
+			result = FilenameUtils.getBaseName(filename);
+			
+			if (StringUtils.isEmpty(result)) {
+				result = "raw";
+			}
+		}
 		
-		return extension;
```

Comment:
I renamed that on purpose since, weirdly enough, that method does not always return the extension. It depends on what you put in, so the result may also just be the filename itself if no extension is found.

Reply:
I agree with @FloBue here but perhaps the Javadoc could be improved to better reflect what this function actually does. It's already been marked as deprecated, which is already a step ahead.

Reply:
@mks-d That's a good idea, I will add/rewrite some information in the Javadoc comment to make clear what to expect as a result.